### ISSUE TITLE
add jest utils 

### ISSUE THREAD

**Issue by IvanIhnatsiuk** [kirillzyusko/react-native-keyboard-controller#110]
**Is your feature request related to a problem? Please describe.**
Hi @kirillzyusko, Thanks a lot for the library. I would like to suggest you implementation of jestUtils, because It would be really great to have possibility to mock keyboard events in jest.

**Describe the solution you'd like**

create a mock implementation for opening/closing keyboard events. In jest it will look like this: 

```ts
const { getByTestId } = render(<SomeComponentWithKeyboardHandler />);

expect(getByTestId("keyboardEventsView").toHaveAnimatedStyle({
 height: 0,
}) // or just simple style if you use Animated from RN core;

RNKeyboardHandler.openKeyboard(state);  // state - height, progress

expect(getByTestId("keyboardEventsView").toHaveAnimatedStyle({
 height: 100,
}) 

RNKeyboardHandler.hideKeyboard(state);  // state - height, progress

expect(getByTestId("keyboardEventsView").toHaveAnimatedStyle({
 height: 0,
}) 

```

**Additional context**
I have some ideas how to implement this logic and we can contact to discuss it :)

**Comment by kirillzyusko** [comment]
Hi @IvanIhnatsiuk 

I wanted to add it, but with the API that you suggested it seems like you will need to change global object in mock:

```ts
RNKeyboardHandler.openKeyboard(state);  // state - height, progress <- seems like here you will change a global value that further is consumed by `useKeyboardAnimation` hook
```

And as far as I remember I experimented with it and it had some problems with tests concurrency. The mock will be created once per file, but since you are changing global object and tests are running in parallel there can be a race condition, where one test will change value and will expect to have that value in next `expect` statement, though it's not always truth, since another test also can change it), for example:

```ts
describe("keyboard unit tests", () => {
  it("should have keyboard height = 0", () => {
    const { getByTestId } = render(<SomeComponentWithKeyboardHandler />);

    RNKeyboardHandler.setState({height: 0, progress: 0});

    expect(getByTestId("keyboardEventsView").toHaveAnimatedStyle({
       height: 0,
    });
  });

  it("should have keyboard height = 100", () => {
    const { getByTestId } = render(<SomeComponentWithKeyboardHandler />);

    RNKeyboardHandler.setState({height: 100, progress: 0.5});

    expect(getByTestId("keyboardEventsView").toHaveAnimatedStyle({
       height: 100,
    });
  });
});
```

Here we have a situation, where two tests are running in parallel, and there might be chances, that `RNKeyboardHandler.setState({height: 100, progress: 0.5});` will be executed straight after `RNKeyboardHandler.setState({height: 0, progress: 0});` and as a result first test will fail.

If you have ideas how to implement such helper/utils that will be able to mock value per tests - let me know, I will be glad to hear that!

<hr />

For now you can simply mock returned values for necessary hooks - you may find examples [here](https://github.com/kirillzyusko/react-native-keyboard-controller/blob/17a9afda05cd10554ab1ed967e8843d6ae078097/example/__tests__/basic-interaction.spec.tsx) and [here](https://github.com/kirillzyusko/react-native-keyboard-controller/blob/17a9afda05cd10554ab1ed967e8843d6ae078097/example/__tests__/reanimated.spec.tsx).

**Comment by kirillzyusko** [comment]
Closed due to inactivity. Feel free to re-open this or open a new issue 😊
